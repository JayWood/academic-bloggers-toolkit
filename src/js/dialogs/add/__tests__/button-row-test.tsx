import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';

import Store from 'stores/ui/add-dialog';
import ButtonRow from '../button-row';

const setup = () => {
    const store = new Store('webpage');
    const onSearchPubmedClick = jest.fn();
    const component = shallow(
        <ButtonRow store={store} onSearchPubmedClick={onSearchPubmedClick} />,
    );
    return {
        component,
        instance: component.instance() as ButtonRow,
        onSearchPubmedClick,
    };
};

describe('<ButtonRow />', () => {
    const BASELINE = toJSON(setup().component);
    it('should match baseline snapshots', () => {
        expect(BASELINE).toMatchSnapshot();
    });
    it('should toggle add manually', () => {
        const { component } = setup();
        const button = component.find('Button').first();
        button.simulate('click');
        expect(toJSON(component)).toMatchDiffSnapshot(BASELINE);
    });
    it('should toggle attach inline', () => {
        const { component } = setup();
        const toggle = component.find('ToggleSwitch');
        toggle.simulate('change');
        expect(toJSON(component)).toMatchDiffSnapshot(BASELINE);
    });
});
