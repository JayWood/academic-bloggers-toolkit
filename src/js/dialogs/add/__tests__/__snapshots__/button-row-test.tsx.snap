// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonRow /> should match snapshots 1`] = `
<div
  className="btn-row"
>
  <Button
    flat={true}
    focusable={true}
    label="Add with Identifier"
    onClick={[Function]}
    type="button"
  />
  <Button
    disabled={true}
    flat={true}
    focusable={true}
    label="Search PubMed"
    onClick={[Function]}
    type="button"
  />
  <span
    className="separator"
  />
  <ToggleSwitch
    checked={false}
    disabled={false}
    onChange={[Function]}
    tooltip={
      Object {
        "position": "left",
        "text": "Insert citation inline",
      }
    }
  />
  <Button
    focusable={true}
    form="add-reference"
    label="Add Reference"
    primary={true}
    type="submit"
  />
  <style
    jsx={true}
  >
    
                        .btn-row {
                            display: flex;
                            padding: 10px;
                            align-items: center;
                            background: #f5f5f5;
                            box-shadow: 0 -1px 0 rgba(16, 22, 26, .10);
                            border-radius: 0 0 2px 2px;
                            margin: 0;
                            justify-content: space-between;
                        }
                        span {
                            border-right: solid 2px rgba(0, 0, 0, 0.12);
                            height: 30px;
                            display: inline-block;
                        }
                    
  </style>
</div>
`;

exports[`<ButtonRow /> should match snapshots 2`] = `
<div
  className="btn-row"
>
  <Button
    flat={true}
    focusable={true}
    label="Add Manually"
    onClick={[Function]}
    type="button"
  />
  <Button
    disabled={false}
    flat={true}
    focusable={true}
    label="Search PubMed"
    onClick={[Function]}
    type="button"
  />
  <span
    className="separator"
  />
  <ToggleSwitch
    checked={false}
    disabled={false}
    onChange={[Function]}
    tooltip={
      Object {
        "position": "left",
        "text": "Insert citation inline",
      }
    }
  />
  <Button
    focusable={true}
    form="add-reference"
    label="Add Reference"
    primary={true}
    type="submit"
  />
  <style
    jsx={true}
  >
    
                        .btn-row {
                            display: flex;
                            padding: 10px;
                            align-items: center;
                            background: #f5f5f5;
                            box-shadow: 0 -1px 0 rgba(16, 22, 26, .10);
                            border-radius: 0 0 2px 2px;
                            margin: 0;
                            justify-content: space-between;
                        }
                        span {
                            border-right: solid 2px rgba(0, 0, 0, 0.12);
                            height: 30px;
                            display: inline-block;
                        }
                    
  </style>
</div>
`;
